src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -59,34 +61,17 @@
  * @since 1.8.0
  */
 public class ArgumentCaptor<T> {
-    
+
     HandyReturnValues handyReturnValues = new HandyReturnValues();
 
-    private final CapturingMatcher<T> capturingMatcher = new CapturingMatcher<T>();
+    private List<Object> arguments;
```

Comment:
let's take a look to Use Case 2: 
if you use two or more matchers that match elements of incompatible type. 

```java

Consumer<List> mock = ...
ArgumentCapture<Array> captor = ...

verfy(mock).consume(captor.captureIf(instanceOf(Array)));
verfy(mock).consume(captor.captureIf(instanceOf(List)));

```

the second matcher will try to add elements instance of `Set` to arguments of type `List`.
This is the reason why I used Object instead of T. What do you think?